app/src/main/java/com/example/composetemplate/presentation/screens/entry_screens/register/RegisterScreen.kt

```diff
@@ -51,19 +55,32 @@ fun RegisterScreen(
     modifier: Modifier = Modifier,
     viewModel: AuthViewModel,
     onLoginClicked: () -> Unit,
-    isRegisterSucceed: SuccessCallback,
+    isRegisterSucceed: (Boolean) -> Unit,
 ) {
+    val signUpResult by viewModel.signInResult.collectAsState()
```

Comment:
Here, for example, you’re using signInResult for signUpResult. It should be something more generic, like FirebaseNetworkResult or something similar.

Reply:
I dissAgree because eventually after the register we need to sign in to get the app. FirebaseNetworkResult is too much generic it doesnt say anything